istioctl/pkg/install/k8sversion/version.go

```diff
@@ -0,0 +1,67 @@
+// Copyright Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sversion
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/version"
+)
+
+const (
+	// MinK8SVersion is the minimum k8s version required to run this version of Istio
+	// https://istio.io/docs/setup/platform-setup/
+	MinK8SVersion = "1.17"
+)
+
+// CheckKubernetesVersion checks if this Istio version is supported in the k8s version
+func CheckKubernetesVersion(versionInfo *version.Info) (bool, error) {
+	v, err := extractKubernetesVersion(versionInfo)
+	if err != nil {
+		return false, err
+	}
+	return parseVersion(MinK8SVersion, 4) <= parseVersion(v, 4), nil
+}
+func extractKubernetesVersion(versionInfo *version.Info) (string, error) {
```

Comment:
Can we add tests for this? with common versions variances like `v1.16.11-gke.5`? I think we have this somewhere already, so maybe it can be reused

Reply:
I think this might be useful ?
https://github.com/istio/istio/blob/master/istioctl/pkg/install/pre-check_test.go#L41